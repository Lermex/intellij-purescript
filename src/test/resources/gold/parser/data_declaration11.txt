Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Foo')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Foo')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('String')
                  PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PsiElement(()('(')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Array')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('User')
                PsiElement())(')')
          PsiElement())(')')