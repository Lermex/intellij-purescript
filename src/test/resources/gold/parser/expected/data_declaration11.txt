Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSDataDeclarationImpl(DATA_DECLARATION)
        PsiElement(PSTokenType.data)('data')
        PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('Foo')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemListImpl(DATA_DECLARATION_ITEM_LIST)
          PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
            PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
              PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('Foo')
            PsiElement(PSTokenType.()('(')
            PSDataDeclarationTypeOnlyImpl(DATA_DECLARATION_TYPE_ONLY)
              PSParenthesisedTypeThenIdListImpl(PARENTHESISED_TYPE_THEN_ID_LIST)
                PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                  PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('String')
              PsiElement(PSTokenType.->)('->')
              PsiElement(PSTokenType.()('(')
              PSParenthesisedTypeThenIdListImpl(PARENTHESISED_TYPE_THEN_ID_LIST)
                PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                  PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('Array')
                  PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                    PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('User')
                    PsiElement(PSTokenType.))(')')
                PsiElement(PSTokenType.))(')')